iroh-blobs/src/util/progress.rs

```diff
@@ -628,3 +720,39 @@ impl<W: AsyncSliceWriter + 'static, F: Fn(u64, usize) -> io::Result<()> + 'stati
         self.0.set_len(size).await
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use super::*;
+
+    #[test]
+    fn get_as_ptr_works() {
+        struct Wrapper(Arc<u64>);
+        let x = Wrapper(Arc::new(1u64));
+        assert_eq!(
+            get_as_ptr(&x).unwrap(),
+            Arc::as_ptr(&x.0) as usize - 2 * std::mem::size_of::<usize>()
+        );
+    }
+
+    #[test]
+    fn get_as_ptr_wrong_use() {
+        struct Wrapper(#[allow(dead_code)] u8);
+        let x = Wrapper(1);
+        assert!(get_as_ptr(&x).is_none());
+    }
+
+    #[test]
+    fn test_sender_is_ptr() {
+        assert_eq!(
+            std::mem::size_of::<usize>(),
+            std::mem::size_of::<flume::Sender<u8>>()
+        );
+        assert_eq!(
+            std::mem::align_of::<usize>(),
+            std::mem::align_of::<flume::Sender<u8>>()
+        );
+    }
```

Comment:
Should we update these tests?

```suggestion
    #[test]
    fn test_sender_is_ptr() {
        assert_eq!(
            std::mem::size_of::<usize>(),
            std::mem::size_of::<async_channel::Sender<u8>>()
        );
        assert_eq!(
            std::mem::align_of::<usize>(),
            std::mem::align_of::<async_channel::Sender<u8>>()
        );
    }
```